drivers/serial/uart_nrfx_uarte.c

```diff
@@ -456,6 +514,7 @@ static int uarte_instance_init(struct device *dev,
 				config->pselrxd);
 
 	if (config->hwfc == NRF_UARTE_HWFC_ENABLED) {
+		data->dev_cfg.flow_ctrl = UART_CFG_FLOW_CTRL_RTS_CTS;
 		nrf_gpio_pin_write(config->pselrts, 1);
 		nrf_gpio_cfg_output(config->pselrts);
```

Comment:
I think now the RTS/CTS pins should be configured when they are only defined in DTS, no matter if flow control is initially enabled or not. And `uarte_nrfx_configure` should check if these pins are defined and only then allow to enable flow control.